### DISCUSSION TITLE
virbtualbox efi mode dose not work

### DISCUSSION THREAD

**Discussion by IAseven** [antonym/netboot.xyz#1453] [General]
**Describe the bug**
![图片](https://github.com/netbootxyz/netboot.xyz/assets/38178039/5c269661-f799-45bb-bd34-b3697d273c67)

used iso file
![图片](https://github.com/netbootxyz/netboot.xyz/assets/38178039/92a4ddc4-514a-4123-a8e3-17ed329ad96e)

**exception**
![图片](https://github.com/netbootxyz/netboot.xyz/assets/38178039/9672d747-dd75-4c87-b94b-394d2a77932e)

**Comment by antonym** [comment]
Try a different virtual network adapter? 

https://github.com/ipxe/ipxe/issues/977#issuecomment-1987283091

**Reply by neitsab** [reply]
I had started writing a bug report, but since the previous one has been turned into a Discussion here's a properly defined test case to make it reproducible:

### VirtualBox/EFI: Booting on the combined ISO works only with the non-Intel network interfaces

- [x] This bug persists when memory is increased beyond 8GB.


**Describe the bug**


Booting on the combined ISO works only with the non-Intel network interfaces, _i.e._ `virtio-net`, `PCnet-PCI II (Am79C970A)` and `PCnet-FAST III (Am79C973)` . Tested with;

* VirtualBox 7.1.4
* Arch Linux x86_64 host
* VM configured in EFI mode
* `PIIX3` and `ICH9` chipset

**To Reproduce**
Steps to reproduce the behavior:
1. Install [VirtualBox 7.1.4 64-bits on Linux](https://www.virtualbox.org/wiki/Linux_Downloads)
2. Create a new VM using 
    - the [x86_64 Combined Legacy and UEFI iPXE Bootloaders](https://boot.netboot.xyz/ipxe/netboot.xyz.iso) ISO image
    - 8 GB of RAM
    - 4 cores
    - **Enable EFI**
    - 20 GB disk
3. Boot the VM: it gets stuck after `file:/autoexec.ipxe... ok`
4. Open the VM Settings, check that Chipset is PIIX3, Network Adapter 1 is using NAT and `Intel PRO/1000 MT Desktop (82540EM)` interface type
4. Change interface to `virtio-net` **and renew the MAC address**
5. Boot the VM: this time it works
6. Shut it down, repeat for all network interface types; none of the Intel ones work, only virtio and PCnet ones
7. Also change the Chipset (under `System` Settings) from PIIX3 to ICH9, to no avail.

Note: switching the interface to Bridge mode makes virtio-net not work anymore.

**Expected behavior**


VM boot on the combined ISO with default settings in EFI mode.

**Screenshots**

Stuck with the default network interface `Intel PRO/1000 MT Desktop (82540EM)`:

![default-intel-pro-desktop](https://github.com/user-attachments/assets/e8dabbb2-c576-458b-a130-b632af793d0b)

Working when changed the interface to virtio-net / PCnet :

![piix3-virtio](https://github.com/user-attachments/assets/064b3fde-c621-4432-88cb-4ea81b44e51e)

I'll try to ping upstream...
